pandas/core/arrays/categorical.py

```diff
@@ -121,9 +121,10 @@ def func(self, other):
                 i = self.categories.get_loc(other)
                 ret = getattr(self._codes, opname)(i)
 
-                # check for NaN in self
-                mask = self._codes == -1
-                ret[mask] = False
+                if opname not in {"eq", "__eq__", "ge", "__ge__", "gt", "__gt__"}:
+                    # check for NaN needed if we are not equal or larger
+                    mask = self._codes == -1
+                    ret[mask] = False
```

Comment:
Do we have a performance test for this?

Reply:
No, no ASV's for this present. I actually can't get ASV to run locally, maybe a Windows issue?

Anyway, I've added I've a ASV test set, but haven't been able to run it myself, unforfunately. Isn't there a web page, where we post ASVs?